src/System.IO.FileSystem/tests/Directory/EnumerableTests.cs

```diff
@@ -31,6 +31,38 @@ public void FileEnumeratorIsThreadSafe()
             {
             }
         }
+        
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void EnumerateDirectories_NonBreakingSpace()
+        {
+            string TestDirectory = GetTestFilePath();
+            DirectoryInfo di = new DirectoryInfo(TestDirectory);
+            di.Create();
+            di.CreateSubdirectory("\u00A0");
+
+            var diQueue = new Queue<DirectoryInfo>();
+            diQueue.Enqueue(di);
+            int count = 0;
+
+            while (diQueue.Count != 0)
+            {
+                foreach (var directory in diQueue.Dequeue().EnumerateDirectories())
+                {
+                    diQueue.Enqueue(directory);
+                }
+
+                count++;
+            }
+
+            Assert.Equal(2, count);
+
+            try
+            {
+                Directory.Delete(TestDirectory, recursive: true);
```

Comment:
This could actually cause other tests to fail as TestDirectory is unique for the class (shared between methods) and created for you.